controllers/kubevirtmachine_controller_test.go

```diff
@@ -268,6 +285,91 @@ var _ = Describe("reconcile a kubevirt machine", func() {
 		Expect(machineContext.KubevirtMachine.Spec.ProviderID).To(BeNil())
 	})
 
+	It("should ensure deletion of KubevirtMachine garbage collects everything successfully", func() {
+		objects := []client.Object{
+			cluster,
+			kubevirtCluster,
+			machine,
+			kubevirtMachine,
+			sshKeySecret,
+			bootstrapSecret,
+			bootstrapUserDataSecret,
+		}
```

Comment:
nice!

once https://github.com/kubernetes-sigs/cluster-api-provider-kubevirt/issues/65 is fixed, we'll extend this unit test to ensure that cleanup can occur when the `cluster` object doesn't exist.